lib/server.js

```diff
@@ -753,17 +753,14 @@ Server.prototype._handle = function _handle(req, res) {
     // run pre() handlers first before routing and running
     if (self.before.length > 0) {
         self._run(req, res, null, self.before, function (err) {
-            // check for return false here - like with the regular handlers,
-            // if false is returned we already sent a response and should stop
-            // processing.
-            if (err === false) {
-                self._finishReqResCycle(req, res);
+            // Like with regular handlers, if we are provided an error, we
+            // should abort the middleware chain and fire after events.
+            if (err !== undefined) {
```

Comment:
This behavior is slightly inconsistent compared to how its implemented in the `use` chain, where we are explicitly checking for `false` or `err`. The result is that calling `return next({})` could trigger an early exit in pre where it wouldn't in use. I don't mind revisiting what nexting with a POJO means, but probably worth tackling in a separate PR?

Reply:
Yup, was on the fence about this. The explicitness bit us in the butt here :smile: so I erred on the side of assuming everything was a cause to abort the chain. Though I guess if someone were to do `next(true)` or `next(null)` this would drop the chain which may not be desirable.